### PR TITLE
Add code for metadata usage in downstream tools

### PR DESCRIPTION
**Pull request by owenrumney**
- create code files that can be used in downstream tools
- add make target to generate

### CODE REVIEW

File: pkg/metadata/aws/apigateway/enable_tracing.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package apigateway
+
+import "github.com/aquasecurity/cloud-metadata/pkg/metadata"
+
+var EnableTracing = metadata.Metadata{
+	ID:          "AVD-AWS-0003",
+	Title:       "API Gateway must have X-Ray tracing enabled",
```
Comment (joshlarsen) [inline pkg/metadata/aws/apigateway/enable_tracing.go:7]: I propose we standardize on an objective statement for the title, and an explanation with a "should" opinionated statement for the description.

```
Title: "API Gateway does not have X-Ray tracing enabled"
```

and

```
Description: "API Gateways should have X-Ray tracing enabled to facilitate end-to-end debugging and analysis of API Gateway requests"
```

File: pkg/metadata/aws/apigateway/enable_tracing.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	Description: "X-Ray tracing enables end-to-end debugging and analysis of all API Gateway HTTP requests.",
+	Impact:      "WIthout full tracing enabled it is difficult to trace the flow of logs",
```
Comment (joshlarsen) [inline pkg/metadata/aws/apigateway/enable_tracing.go:9]: `Without`

File: pkg/metadata/aws/apigateway/no_public_access.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package apigateway
+
+import "github.com/aquasecurity/cloud-metadata/pkg/metadata"
+
+var NoPublicAccess = metadata.Metadata{
+	ID:          "AVD-AWS-0004",
+	Title:       "No public access to API Gateway methods",
```
Comment (joshlarsen) [inline pkg/metadata/aws/apigateway/no_public_access.go:7]: Prefer this format.

File: pkg/metadata/aws/apigateway/no_public_access.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	Description: "API Gateway methods should be protected by authorization or api key. OPTION verb calls can be used without authorization",
```
Comment (joshlarsen) [inline pkg/metadata/aws/apigateway/no_public_access.go:8]: Prefer this format.

File: pkg/metadata/aws/apigateway/use_secure_tls_policy.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package apigateway
+
+import "github.com/aquasecurity/cloud-metadata/pkg/metadata"
+
+var UseSecureTlsPolicy = metadata.Metadata{
+	ID:          "AVD-AWS-0005",
+	Title:       "API Gateway domain name uses outdated SSL/TLS protocols.",
+	Description: "You should not use outdated/insecure TLS versions for encryption. You should be using TLS v1.2+.",
```
Comment (joshlarsen) [inline pkg/metadata/aws/apigateway/use_secure_tls_policy.go:8]: Avoid second person perspective.

Prefer

```
Outdated/insecure TLS encryption should not be used. Any version prior to TLS v1.2 is considered insecure and outdated.
```

over

```
You should not use outdated/insecure TLS versions for encryption. You should be using TLS v1.2+.
```

File: pkg/metadata/aws/athena/enable_at_rest_encryption.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package athena
+
+import "github.com/aquasecurity/cloud-metadata/pkg/metadata"
+
+var EnableAtRestEncryption = metadata.Metadata{
+	ID:          "AVD-AWS-0006",
+	Title:       "Athena databases and workgroup configurations are created unencrypted at rest by default, they should be encrypted",
```
Comment (joshlarsen) [inline pkg/metadata/aws/athena/enable_at_rest_encryption.go:7]: Prefer

```
Unencrypted Athena database or workspace.
```

File: pkg/metadata/aws/athena/enable_at_rest_encryption.go [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+	Description: "Athena databases and workspace result sets should be encrypted at rests. These databases and query sets are generally derived from data in S3 buckets and should have the same level of at rest protection.",
```
Comment (joshlarsen) [inline pkg/metadata/aws/athena/enable_at_rest_encryption.go:8]:
```
"Though not encrypted by default, Athena databases and workspace result sets should be encrypted at rest. These databases and query sets are generally derived from data in S3 buckets and should have the same level of at-rest protection."
```

**Review by joshlarsen — COMMENTED** [review]
Spot checked and left a few unsolicited suggestions.

**Comment by owenrumney** [comment]
Managing the descriptions and titles here - 
https://aquasecurity-my.sharepoint.com/:x:/g/personal/owen_rumney_aquasec_com/EX9vHs9i0XlJouKxPO3zO24BQooz85dZyEOmaYOFOkx1DQ?e=dsV1XQ

**Review by liamg — APPROVED** [review]

**Pull request merged**